python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -267,18 +267,25 @@ def _update(self):
                     self.load_metrics.mark_active(
                         self.provider.internal_ip(node_id))
                 else:
-                    logger.error(f"StandardAutoscaler: {node_id}: Terminating "
-                                 "failed to setup/initialize node.")
+                    nodes_to_terminate.append(node_id)
+                    self.num_failed_updates[node_id] += 1
+                del self.updaters[node_id]
+            if nodes_to_terminate:
+                # Some nodes in nodes_to_terminate may have been terminated
+                # during an update. Only terminate currently active nodes.
+                active_nodes = self.workers()
+                active_nodes_to_terminate = [
+                    node for node in nodes_to_terminate if node in active_nodes
+                ]
+                for node_id in active_nodes_to_terminate:
+                    logger.error(f"StandardAutoscaler: {node_id}: Terminating."
+                                 " Failed to setup/initialize node.")
                     self.event_summarizer.add(
                         "Removing {} nodes of type " +
                         self._get_node_type(node_id) + " (launch failed).",
                         quantity=1,
                         aggregate=operator.add)
-                    nodes_to_terminate.append(node_id)
-                    self.num_failed_updates[node_id] += 1
-                del self.updaters[node_id]
-            if nodes_to_terminate:
-                self.provider.terminate_nodes(nodes_to_terminate)
+                self.provider.terminate_nodes(active_nodes_to_terminate)
```

Comment:
The diff is hard to read -- I moved the `logger.error` / `self.event_summarizer` part into a for loop under the `if nodes_to_terminate` block.